Buildings/Controls/OBC/ASHRAE/PrimarySystem/BoilerPlant/Controller.mo

```diff
@@ -0,0 +1,2260 @@
+
+  parameter Real oneReqLimHig_priPum(
+    final unit="K",
+    displayUnit="K",
+    final quantity="TemperatureDifference") = 1
+    "Higher limit of hysteresis loop sending one request"
+    annotation (Dialog(tab="Primary pump control parameters",
+      group="Temperature-based speed regulation",
+      enable = speConTypPri == Buildings.Controls.OBC.ASHRAE.PrimarySystem.BoilerPlant.Types.PrimaryPumpSpeedControlTypes.temperature));
+
+  parameter Real k_priPum(
+    final unit="1",
+    displayUnit="1",
+    final min=0) = 1
+    "Gain of controller"
+    annotation (Dialog(tab="Primary pump control parameters", group="PID parameters"));
+
+  parameter Real Ti_priPum(
+    final unit="s",
+    displayUnit="s",
+    final quantity="time",
+    final min=0) = 0.5
+    "Time constant of integrator block"
+    annotation (Dialog(tab="Primary pump control parameters", group="PID parameters"));
+
+  parameter Real Td_priPum(
+    final unit="s",
+    displayUnit="s",
+    final quantity="time",
+    final min=0) = 0.1
+    "Time constant of derivative block"
+    annotation (Dialog(tab="Primary pump control parameters", group="PID parameters"));
+
+  parameter Real TRetSet(
+    final unit="K",
+    displayUnit="K",
+    final quantity="ThermodynamicTemperature") = 333.15
+    "Minimum hot water return temperature for optimal non-condensing boiler performance"
+    annotation(Dialog(tab="Condensation control parameters"));
+
+  parameter Real TRetMinAll(
+    final unit="K",
+    displayUnit="K",
+    final quantity="ThermodynamicTemperature") = 330.35
+    "Minimum allowed hot water return temperature for non-condensing boiler"
+    annotation(Dialog(tab="Condensation control parameters"));
+
+  parameter Real minSecPumSpe(
+    final unit="1",
+    displayUnit="1",
+    final min=0,
+    final max=1) = 0
+    "Minimum secondary pump speed"
+    annotation(Dialog(tab="Condensation control parameters"));
+
+  parameter Real minPriPumSpeSta[nSta](
+    final unit=fill("1",nSta),
+    displayUnit=fill("1",nSta),
+    final min=fill(0,nSta),
+    final max=fill(1,nSta))
+    "Vector of minimum primary pump speed for each stage"
+    annotation(Dialog(group="Boiler plant configuration parameters"));
+
+  parameter Real minPumSpeSec(
+    final unit="1",
+    displayUnit="1",
+    final min=0,
+    final max=maxPumSpeSec) = 0.1
+    "Minimum pump speed"
+    annotation (Dialog(tab="Secondary pump control parameters",
+      group="General parameters",
+      enable=have_varSecPum));
+
+  parameter Real maxPumSpeSec(
+    final unit="1",
+    displayUnit="1",
+    final min=minPumSpeSec,
+    final max=1) = 1
+    "Maximum pump speed"
+    annotation (Dialog(tab="Secondary pump control parameters",
+      group="General parameters",
+      enable=have_varSecPum));
+
+  parameter Real VHotWatSec_flow_nominal(
+    final unit="m3/s",
+    displayUnit="m3/s",
+    final quantity="VolumeFlowRate") = 0.5
+    "Secondary loop design hot water flow rate"
```

Comment:
Missing enable condition: only required for primary-secondary plants with a flow meter in the secondary loop, see G36 3.1.8.4.